independent-projects/arc/runtime/src/main/java/io/quarkus/arc/InjectableBean.java

```diff
@@ -162,6 +162,13 @@ default Class<?> getImplementationClass() {
         return getBeanClass();
     }
 
+    /**
+     * Checks whether this (synthetic) bean is active. To signal that this bean is <em>not</em> active,
+     * this method should throw an exception. If this method does not throw, this bean is considered active.
+     */
+    default void checkActive() {
```

Comment:
Why do we need this method? Could we actually do the check when a bean is created?

Reply:
Good point, we can generate the bytecode directly instead of going through this method. Will remove.

Reply:
Done.